src/Plugin/formatter/FormatterJson.php

```diff
@@ -98,13 +117,17 @@ protected function extractFieldValues($data) {
       }
       $output[$public_field_name] = $value;
     }
+    if ($this->isCacheEnabled($data)) {
```

Comment:
not very beautiful, but maybe have the main method broken to three parts? -- implementors will have to deal only with the middle part